include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -161,36 +169,29 @@ def FModuleLike : OpInterface<"FModuleLike"> {
     }]>,
 
     InterfaceMethod<"Get a port symbol attribute",
-    "StringAttr", "getPortSymbolAttr", (ins "size_t":$portIndex), [{}], [{
+    "InnerSymAttr", "getPortSymbolAttr", (ins "size_t":$portIndex), [{}], [{
       auto syms = $_op.getPortSymbols();
-      if (syms.empty())
-        return StringAttr::get($_op.getContext(), "");
-      return syms[portIndex].template cast<StringAttr>();
+      if (syms.empty() ||
+          syms[portIndex].template cast<InnerSymAttr>().isSymbolInvalid())
```

Comment:
Can we reject invalid symbols at the verifier level?

Reply:
Unfortunately, invalid just means symbol name is empty string. And as we discussed we need empty string symbol names because we can't have null attributes. So, we need to add `""` as symbol name and then use `isSymbolInvalid` to ignore them.